src/Inventory/Asset/Software.php

```diff
@@ -251,6 +254,7 @@ public function handle()
         }
 
         $db_software = [];
+        $db_software_wo_version = [];
```

Comment:
This is a bit hard to read.

Wasn't it possible to store only in `$db_software`, using always key generated without version, and then adapt tests ?
```
$key = $this->getFullCompareKey((object)$data, false);
$db_software[$key] = [
    'id'          => $softid,
    'version' => $data['version'],
    'name'    => $data['name'],
];
....
if (isset($db_software[$key]) && $db_software[$key]['version'] === $val->version) {
    ....
} else {
    ....
}
```